ts/features/bonus/common/components/BonusInformationComponent.tsx

```diff
@@ -90,6 +90,59 @@ const styles = StyleSheet.create({
 const loadingOpacity = 0.9;
 // for long content markdown computed height should be not enough
 const extraMarkdownBodyHeight = 20;
+
+// TODO get the tos footer from props
+const getTosFooter = (
+  maybeBonusTos: Option<string>,
+  maybeRegulationUrl: Option<{ url: string; name: string }>,
+  handleModalPress: (tos: string) => void
+) => {
+  // if tos and regulation url is defined return a markdown footer including both links reference (BPD)
+  if (maybeBonusTos.isSome() && maybeRegulationUrl.isSome()) {
```

Comment:
You check 2 times if "maybeBonusTos.isSome()".

You can:

```
if(maybeBonusTos.isSome()) {
maybeRegulationUrl.isSome() ? return X : return Y
}
return null
```

TIP: You can change the style and use chain/fold/map and so on.